lib/google/ads/google_ads/google_ads_client.rb

```diff
@@ -96,6 +97,25 @@ def configure(&block)
           yield @config
         end
 
+        def configure_using_environment_variables
+          # Client library-generic variables
+          @config.client_id = ENV.fetch("GOOGLE_ADS_CLIENT_ID", @config.client_id)
+          @config.client_secret = ENV.fetch("GOOGLE_ADS_CLIENT_SECRET", @config.client_secret)
+          @config.keyfile = ENV.fetch("GOOGLE_ADS_JSON_KEY_FILE_PATH", @config.keyfile)
+          @config.impersonate = ENV.fetch("GOOGLE_ADS_IMPERSONATED_EMAIL", @config.impersonate)
+          @config.developer_token = ENV.fetch("GOOGLE_ADS_DEVELOPER_TOKEN", @config.developer_token)
+          @config.login_customer_id = ENV.fetch("GOOGLE_ADS_LOGIN_CUSTOMER_ID", @config.login_customer_id)
+          @config.linked_customer_id = ENV.fetch("GOOGLE_ADS_LINKED_CUSTOMER_ID", @config.linked_customer_id)
+
+          # Client library-specific variables
+          @config.treat_deprecation_warnings_as_errors = ENV.fetch("GOOGLE_ADS_RUBY_TREAT_DEPRECATION_WARNINGS_AS_ERRORS", @config.treat_deprecation_warnings_as_errors)
+          @config.warn_on_all_deprecations = ENV.fetch("GOOGLE_ADS_RUBY_WARN_ON_ALL_DEPRECATIONS", @config.warn_on_all_deprecations)
+          @config.authentication = ENV.fetch("GOOGLE_ADS_RUBY_AUTHENTICATION", @config.authentication)
+          @config.log_level = ENV.fetch("GOOGLE_ADS_RUBY_LOG_LEVEL", @config.log_level)
+          @config.log_target = ENV.fetch("GOOGLE_ADS_RUBY_LOG_TARGET", @config.log_target)
+          @config.logger = ENV.fetch("GOOGLE_ADS_RUBY_LOGGER", @config.logger)
```

Comment:
Same here. You can't specify this one as a string; it must be a Ruby object. So don't include it here.

Reply:
Removed the ones that don't work with strings.